src/components/common/YourCompany/EntityName.vue

```diff
@@ -399,29 +399,27 @@ export default class EntityName extends Mixins(CommonMixin, NameRequestMixin) {
   /** The current options for name changes. */
   get correctNameChoices (): Array<CorrectNameOptions> {
     // safety check
-    const changeData = this.getResource.changeData
-    if (!changeData) return []
-
-    const filterdOptionByAuthAction = IsAuthorized(AuthorizedActions.ADD_ENTITY_NO_AUTHENTICATION)
-      ? CorrectNameOptions.CORRECT_NEW_NR
-      : CorrectNameOptions.CORRECT_NEW_NR_STAFF
-
-    // Filter out the option based on authorized action
-    let filteredOptions = changeData.correctNameOptions.filter(
-      option => option !== filterdOptionByAuthAction
-    )
+    let correctNameOptions = this.getResource.changeData?.correctNameOptions
+    if (!correctNameOptions) return []
+
+    // check for (non-staff) correct-new-nr option
+    const index = correctNameOptions.indexOf(CorrectNameOptions.CORRECT_NEW_NR)
+    if (index >= 0 && IsAuthorized(AuthorizedActions.ADD_ENTITY_NO_AUTHENTICATION)) {
+      // if user is authorized to add an entity without authentication
+      // then replace non-staff option with staff option
+      correctNameOptions.splice(index, 1, CorrectNameOptions.CORRECT_NEW_NR_STAFF)
+    }
```

Comment:
I decided to change the code instead of adding CORRECT_NEW_NR_STAFF to about a dozen resource files.